lib/gitspindle/github.py

```diff
@@ -1115,7 +1115,14 @@ def pull_request(self, opts):
         if not src:
             src = self.gitm('rev-parse', '--abbrev-ref', 'HEAD').stdout.strip()
         if not dst:
-            dst = parent.default_branch
+            tracking_remote = None
+            tracking_branch = self.git('rev-parse', '--abbrev-ref', '%s@{u}' % src).stdout.strip()
+            if '/' in tracking_branch:
+                tracking_remote, tracking_branch = tracking_branch.split('/', 1)
+            if (parent == repo and tracking_remote == 'origin') or (parent != repo and tracking_remote == 'upstream'):
+                dst = tracking_branch
```

Comment:
Not sure if hardcoding origin/upstream is sane here. Maybe we should verify whether they point to the right repo?